zcash_client_sqlite/src/lib.rs

```diff
@@ -490,58 +488,40 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
                         }
                     }
 
-                    wallet::sapling::put_received_note(&wdb.conn.0, output, tx_ref)?;
+                    wallet::sapling::put_received_note(wdb.conn.0, output, tx_ref)?;
                 }
             }
-        }
 
-        // If any of the utxos spent in the transaction are ours, mark them as spent.
-        #[cfg(feature = "transparent-inputs")]
-        for txin in d_tx
-            .tx
-            .transparent_bundle()
-            .iter()
-            .flat_map(|b| b.vin.iter())
-        {
-            wallet::mark_transparent_utxo_spent(&wdb.conn.0, tx_ref, &txin.prevout)?;
-        }
+            // If any of the utxos spent in the transaction are ours, mark them as spent.
+            #[cfg(feature = "transparent-inputs")]
+            for txin in d_tx.tx.transparent_bundle().iter().flat_map(|b| b.vin.iter()) {
+                wallet::mark_transparent_utxo_spent(wdb.conn.0, tx_ref, &txin.prevout)?;
+            }
 
-        // If we have some transparent outputs:
-        if !d_tx
-            .tx
-            .transparent_bundle()
-            .iter()
-            .any(|b| b.vout.is_empty())
-        {
-            let nullifiers = wdb.get_sapling_nullifiers(data_api::NullifierQuery::All)?;
-            // If the transaction contains shielded spends from our wallet, we will store z->t
-            // transactions we observe in the same way they would be stored by
-            // create_spend_to_address.
-            if let Some((account_id, _)) = nullifiers.iter().find(|(_, nf)| {
-                d_tx.tx
-                    .sapling_bundle()
-                    .iter()
-                    .flat_map(|b| b.shielded_spends().iter())
-                    .any(|input| nf == input.nullifier())
-            }) {
-                for (output_index, txout) in d_tx
-                    .tx
-                    .transparent_bundle()
-                    .iter()
-                    .flat_map(|b| b.vout.iter())
-                    .enumerate()
-                {
-                    if let Some(address) = txout.recipient_address() {
-                        wallet::put_sent_output(
-                            &wdb.conn.0,
-                            &wdb.params,
-                            *account_id,
-                            tx_ref,
-                            output_index,
-                            &Recipient::Transparent(address),
-                            txout.value,
-                            None,
-                        )?;
+            // If we have some transparent outputs:
+            if !d_tx.tx.transparent_bundle().iter().any(|b| b.vout.is_empty()) {
```

Comment:
This asks if it is not the case that any of the vouts are empty, i.e. if all of the vouts are non-empty. That doesn't seem to correspond to the comment?

Reply:
I really wish that `Option` defined `exists` and `forall` directly. But you're correct, I think this should be 
```suggestion
            if d_tx.tx.transparent_bundle().iter().any(|b| !b.vout.is_empty()) {
```